controller/machine/controller.go

```diff
@@ -109,6 +111,14 @@ func (m *Lifecycle) Remove(obj *v3.Machine) (*v3.Machine, error) {
 	if obj.Status.MachineTemplateSpec == nil {
 		return obj, nil
 	}
+	singleEtcdOrControl, err := checkSingleNode(obj, m.clusters, m.machineClient)
+	if err != nil {
+		return obj, err
+	}
+
+	if singleEtcdOrControl {
+		return obj, nil
```

Comment:
we should log some warning here to indicate that no action was taken because it is the last node